packages/amplify-graphql-model-transformer/src/graphql-model-transformer.ts

```diff
@@ -323,9 +324,31 @@ export class ModelTransformer extends TransformerModelBase implements Transforme
   };
 
   generateResolvers = (context: TransformerContextProvider): void => {
+    const dataSourceMapping = {};
     this.resourceGeneratorMap.forEach((generator) => {
       generator.generateResources(context);
+      Object.assign(
+        dataSourceMapping,
+        Object.fromEntries(
+          Object.entries(generator.getDatasourceMap())
+            .filter(([, datasource]) => datasource.ds.type === 'AMAZON_DYNAMODB')
+            .map(([modelName, datasource]) => {
+              if (datasource.ds.dynamoDbConfig && !cdk.isResolvableObject(datasource.ds.dynamoDbConfig)) {
+                return [modelName, datasource.ds.dynamoDbConfig.tableName];
+              }
+              throw new Error(`Unable to construct data source mapping. Could not resolve table name for ${modelName}`);
+            }),
+        ),
+      );
     });
+    if (context.transformParameters.enableGen2Migration && context.transformParameters.enableTransformerCfnOutputs) {
+      const { scope } = context.stackManager;
+      new cdk.CfnOutput(cdk.Stack.of(scope), 'DataSourceMappingOutput', {
+        value: cdk.Stack.of(scope).toJsonString(dataSourceMapping),
+        description: 'Mapping of model name to data source table name.',
+        exportName: cdk.Fn.join(':', [cdk.Aws.STACK_NAME, 'DataSourceMappingOutput']),
```

Comment:
Let's review the name of this output, since it's a new public API (my first thought is that it's too broad--it's specifically a mapping of Amplify managed tables, and possibly just of imported tables?